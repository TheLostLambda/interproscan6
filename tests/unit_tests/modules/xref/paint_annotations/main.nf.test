nextflow_process {

    name "Test Process PAINT_ANNOTATIONS"
    script "interproscan/modules/xrefs/paint_annotations/main.nf"
    process "PAINT_ANNOTATIONS"

    stage {
        copy "tests/unit_tests/test_inputs/xrefs/paint_annotations/PAINT_Annotations"
        copy "tests/unit_tests/test_inputs/xrefs/paint_annotations/PTHR11228.json"
        copy "tests/unit_tests/test_inputs/xrefs/paint_annotations/PTHR43780.json"
        copy "interproscan/scripts/xrefs/paint_annotations.py"
        copy "tests/unit_tests/test_inputs/xrefs/paint_annotations/entries_matches.json"
    }

    test("Should run without failures") {

        when {
            params {
            }
            process {
                """
                input[0] = file("meta/tests/unit_tests/test_inputs/xrefs/paint_annotations/entries_matches.json")
                input[1] = Channel.fromPath("../../../meta/tests/unit_tests/test_inputs/xrefs/paint_annotations/*")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
