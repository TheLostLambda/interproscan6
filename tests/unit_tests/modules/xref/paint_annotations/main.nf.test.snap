{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-09-08T20:00:39.016894"
    }
}